p2p/upgrade.go

```diff
@@ -75,35 +93,36 @@ func Upgrade(h host.Host, opts ...Opt) (*Host, error) {
 		opt(fh)
 	}
 	cfg := fh.cfg
-	var err error
-	fh.PubSub, err = pubsub.New(fh.ctx, fh.logger, h, pubsub.Config{
+	bootnode, err := isBootnode(h, cfg.Bootnodes)
```

Comment:
this probably should be a part of the user provided config. technically bootnodes list should not include its own node, e.g. it requires preparing full libp2p entry before starting a bootnode. i don't do that in system tests, and *i think* we don't do that for public testnets as well

Reply:
the deployment follows this process for bootnode:
1. launch a node
2. node will generate p2p private key
3. request p2p public key from api
4. create libp2p entry from public key + ip/port
5. put that entry into config for nodes that won't serve as bootnodes

so as you see bootnode will not have its own entry in the config

Reply:
thanks for the explanation.